spec/factories/generic_object_definition.rb

```diff
@@ -2,4 +2,14 @@
   factory :generic_object_definition do
     sequence(:name) { |n| "generic_object_definition_#{seq_padded_for_sorting(n)}" }
   end
+
+  trait :with_methods_attributes_associations do
+    properties do
+      {
+        :methods      => %w(add_vms remove_vms),
+        :attributes   => { 'powered_on' => 'boolean', 'widget' => 'string' },
+        :associations => { 'vms' => 'Vm', 'services' => 'Service' }
+      }
+    end
+  end
```

Comment:
Using a trait instead of defining another factory like we always do 😍 

This `trait` call should be within the `factory` block, no?